### DISCUSSION TITLE
storeToRefs with Typescript

### DISCUSSION THREAD

**Discussion by 209** [vuejs/pinia#956] [Help and Questions] [answered]
```ts
export const useLocationStore = defineStore('location', {
  state():Location {
    const myCvStore = useUserStore()
    const { location } = storeToRefs(myCvStore)

    return location
  },
})

```
I have ts error:
> TS2740: Type 'Ref... is missing the following properties from type 'Location'

because **location** is Ref<Location>

need help for understanding)

**Comment by andorfermichael** [comment]
I am also facing this issue. :(

**Comment by posva** [comment, marked as answer]
Why are you returning a state property of a different store in that store?

As for the usage, the return of `state` should be a plain object with properties `return { location }` if anything

**Comment by wesleylether** [comment]
@209  the storeToRefs is a tool to use refs outside your store.
So when you use a store with `const store = useMyStore()`
When in your store are refs you use them with that wrapper to preserve its refs trough destructuring.
`const { refName } = storeToRefs(store)`

See documentation:
https://pinia.vuejs.org/core-concepts/#Destructuring-from-a-Store
